File: src/model/notification_model.ts
export interface NotificationModel {
  id: string;
  updated_at: string;
  subject: Subject;
  repository: Repository;
}

interface Subject {
  title: string;
  url: string;
  type: string;
}

interface Repository {
  full_name: string;
  owner: Owner;
}

interface Owner {
  avatar_url: string;
}

File: src-tauri/src/notification_model.rs
pub mod model {
    use serde::{Deserialize, Serialize};
    
    #[derive(Serialize, Deserialize)]
    pub struct Notification {
        pub id: String,
        pub updated_at: String,
        pub subject: Subject,
        pub repository: Repository,
    }

    #[derive(Serialize, Deserialize)]
    pub struct Subject {
        pub title: String,
        pub url: String,
        #[serde(rename = "type")]
        pub subject_type: String,
    }

    #[derive(Serialize, Deserialize)]
    pub struct Repository {
        pub full_name: String,
        pub owner: Owner,
    }

    #[derive(Serialize, Deserialize)]
    pub struct Owner {
        pub avatar_url: String,
    }
}
